File: contests/lc/1018.BinaryPrefixDivisibleBy5/solution.py
from typing import List


def prefixesDivBy5(nums: List[int]) -> List[bool]:
    """
    Return a list where each element indicates whether the corresponding
    binary prefix of nums is divisible by 5.
    """
    result: List[bool] = []
    mod = 0

    for bit in nums:
        mod = ((mod << 1) + bit) % 5
        result.append(mod == 0)

    return result


if __name__ == "__main__":
    sample = [0, 1, 1]
    print(prefixesDivBy5(sample))

File: contests/lc/1018.BinaryPrefixDivisibleBy5/solution.go
package main

// prefixesDivBy5 returns a boolean slice where each element indicates whether
// the corresponding binary prefix of nums is divisible by 5. We keep only the
// value modulo 5 while iterating to avoid overflow and unnecessary large ints.
func prefixesDivBy5(nums []int) []bool {
	result := make([]bool, len(nums))
	mod := 0

	for i, bit := range nums {
		mod = ((mod << 1) + bit) % 5
		result[i] = mod == 0
	}

	return result
}
